app/controllers/Whereuat.scala

```diff
@@ -110,18 +105,24 @@ class Whereuat extends Controller {
   // then add them to the clients collection if it does.
   def createAccount = Action(parse.json) { request =>
     request.body.validate(createReads).map {
-      case (phone, gcm, vCode) =>
-        if (isVerified(phone, vCode)) {
+      case (phone, gcm, vCode, os) =>
+        val verified = isVerified(phone, vCode)
+        val valid_os = isValidOs(os)
+        if (verified && valid_os) {
           val query = MongoDBObject("phone-#" -> phone)
-          val client = MongoDBObject("gcm-token" -> gcm, "phone-#" -> phone)
+          val client = MongoDBObject("gcm-token" -> gcm, 
+                                     "phone-#" -> phone, 
+                                     "client-os" -> os)
           // Update with an upsert rather than insert in order to handle a
           // client needing to create their account.
           db("clients").update(query, client, upsert=true)
           db("verifiers").remove(query)
           Ok(s"Created account for phone number $phone\n")
-        } else {
+        } else if (!verified) {
           InternalServerError("VERIFICATION ERROR: Verification codes do not " +
                               "match\n")
+        } else {
```

Comment:
Just for reference, on the android code we had something similar like this but we put an explicit `else if (!valid_os)` so you could quickly see that that's what gets you into this code path. I don't really care either way but I just thought I'd make the observation.